akka-actor/src/main/scala/akka/util/ByteString.scala

```diff
@@ -274,8 +286,44 @@ object ByteString {
     def asByteBuffers: scala.collection.immutable.Iterable[ByteBuffer] = bytestrings map { _.asByteBuffer }
 
     def decodeString(charset: String): String = compact.decodeString(charset)
+
+    private[akka] def writeToOutputStream(os: OutputStream): Unit =
+      bytestrings.foreach(_.writeToOutputStream(os))
```

Comment:
maybe we can leave ByteStrings sparse, but compact all ByteString1 (and components of ByteStrings) to ByteString1C